### PR TITLE
Ensure claims from 3rd party ID provider are pulled through.

### PR DESCRIPTION
**Pull request by michaelvidal24**
Fix up swagger URL when not running from the top level website.
Add config to disable Windows Auth when running in IIS.

---
This change is [Reviewable](https://reviewable.io/reviews/healthcatalyst/fabric.identity/153)

### CODE REVIEW

**Comment by michaelvidal24** [comment]
@kpaul, @bsmithvt - Can you guys take a look at my latest updates to this PR? I had to switch to persisting in the db to handle the scenario where the server restarted (thus clearing the ConcurrentDictionary) but the client still had a valid token and attempted to call the userinfo endpoint.

File: Fabric.Identity.API/Persistence/SqlServer/Stores/SqlServerUserStore.cs [hunk 1]
```diff
@@ -66,9 +58,8 @@ public SqlServerUserStore(IIdentityDbContext identityDbContext)
         public async Task<User> AddUserAsync(User user)
         {
             var userEntity = user.ToEntity();
-
+            userEntity.Claims = user.Claims.Select(c => new UserClaim {Type = c.Type, Value = c.Value}).ToList();
```
Comment (kpaul) [inline Fabric.Identity.API/Persistence/SqlServer/Stores/SqlServerUserStore.cs]: you could move this line into the ToEntity extension method that does not take an existing entity as a parameter.

File: Fabric.Identity.API/Persistence/SqlServer/Stores/SqlServerUserStore.cs [hunk 2]
```diff
@@ -90,19 +81,21 @@ public void UpdateUser(User user)
...
+                if (userClaim != null)
+                {
+                    userClaim.Value = claim.Value;
+                }
+                else
+                {
+                    existingUser.Claims.Add(new UserClaim{Type = claim.Type, Value = claim.Value});
+                }
+            }
+
```
Comment (kpaul) [inline Fabric.Identity.API/Persistence/SqlServer/Stores/SqlServerUserStore.cs]: you could move this line into the ToEntity extension method that accepts an existing entity as a parameter

File: Fabric.Identity.API/Persistence/SqlServer/Stores/SqlServerUserStore.cs [hunk 3]
```diff
@@ -90,19 +81,21 @@ public void UpdateUser(User user)
 
             user.ToEntity(existingUser);
```
Comment (bsmithvt) [inline Fabric.Identity.API/Persistence/SqlServer/Stores/SqlServerUserStore.cs]: Is there a case where claims can be removed?

Comment (michaelvidal24) [reply Fabric.Identity.API/Persistence/SqlServer/Stores/SqlServerUserStore.cs]: Good point, i've updated the code to handle this situation, and also added tests to cover it.

File: Fabric.Identity.API/Persistence/SqlServer/Stores/SqlServerUserStore.cs (hunk 1 above)
Comment (michaelvidal24) [reply Fabric.Identity.API/Persistence/SqlServer/Stores/SqlServerUserStore.cs]: Good suggestions, I done that for both this comment and the one below.

**Comment by bsmithvt** [comment]
Should Claim Tyoe and Value comparisons be case-insensitive?

**Comment by michaelvidal24** [comment]
They should be case sensitive in this case as certain claims are case sensitive.

**Pull request merged**